tests/core/data/mock_couchbase.py

```diff
@@ -0,0 +1,59 @@
+class MockCouchbaseCollection:
+    def __init__(self):
+        self.documents = {}
+
+    def insert(self, doc_id, document):
+        if doc_id in self.documents:
+            raise Exception(f"Document with ID {doc_id} already exists.")
+        self.documents[doc_id] = document
+
+    def upsert(self, doc_id, document):
+        self.documents[doc_id] = document
+
+    def get(self, doc_id):
+        if doc_id not in self.documents:
+            raise Exception(f"Document with ID {doc_id} not found.")
+        return self.documents[doc_id]
+
+    def remove(self, doc_id):
+        if doc_id not in self.documents:
+            raise Exception(f"Document with ID {doc_id} not found.")
+        del self.documents[doc_id]
```

Comment:
As my understanding, the MockCouchbaseCollection is essentially a dictionary right?